File: pe-core/src/main/java/org/saar/core/common/terrain/height/NoiseHeightGenerator.java
package org.saar.core.common.terrain.height;

import org.saar.maths.noise.Noise2f;

public class NoiseHeightGenerator implements HeightGenerator {

    private final Noise2f noise;

    public NoiseHeightGenerator(Noise2f noise) {
        this.noise = noise;
    }

    @Override
    public float generateHeight(float x, float z) {
        return this.noise.noise(x, z);
    }
}

File: pe-core/src/main/java/org/saar/core/common/terrain/height/NoiseHeightGenerator.kt
package org.saar.core.common.terrain.height

import org.saar.maths.noise.Noise2f
import org.saar.maths.noise.Noise3f

class NoiseHeightGenerator(private val noise: Noise3f) : HeightGenerator {

    constructor(noise: Noise2f) : this({ x, _, z -> noise.noise(x, z) })

    override fun generate(x: Float, y: Float, z: Float) = this.noise.noise(x, y, z)

}